in_toto/models/metadata.py

```diff
@@ -23,20 +23,86 @@
 """
 
 import attr
+import copy
 import json
+from typing import Optional, List, Union
 
 import securesystemslib.keys
 import securesystemslib.formats
 import securesystemslib.exceptions
 import securesystemslib.gpg.functions
+from securesystemslib.key import Key
+from securesystemslib.metadata import Envelope as SSlibEnvelope
+from securesystemslib.signer import GPGSignature, Signature, Signer
+from securesystemslib.serialization import (BaseDeserializer, BaseSerializer,
+    JSONDeserializer, JSONSerializer, SerializationMixin)
+from in_toto.exceptions import InvalidMetadata
 
 from in_toto.models.common import ValidationMixin
 from in_toto.models.link import Link
 from in_toto.models.layout import Layout
-from in_toto.exceptions import SignatureVerificationError
+
+
+class AnyMetadataDeserializer(JSONDeserializer):
+  """Deserialize bytes into Metablock / DSSE Envelope."""
+
+  def deserialize(self, raw_data: bytes) -> Union["Envelope", "Metablock"]:
+    """Deserialize JSON bytes into Metablock or Envelope as per ITE-5. """
+
+    data = super().deserialize(raw_data)
+
+    if "payload" in data:
+      if data.get("payloadType") == "application/vnd.in-toto+json":
```

Comment:
Can you make `"application/vnd.in-toto+json"` a constant either in the metadata module or on the `Envelope` class and also use that constant in `create_envelope`?